seatunnel-core/seatunnel-core-spark/src/main/java/org/apache/seatunnel/SparkStarter.java

```diff
@@ -293,6 +296,24 @@ protected void appendAppJar(List<String> commands) {
         commands.add(Common.appLibDir().resolve("seatunnel-core-spark.jar").toString());
     }
 
+    /**
+     * change file location to get config file on yarn cluster mode
+     */
+    protected void changeFileLocation() {
+        if (DeployMode.CLUSTER.getName().equals(this.commandArgs.getDeployMode().getName())) {
+            String regEx = ".+/(.+)$";
+            Pattern p = Pattern.compile(regEx);
+            for (int i = 1; i < args.length; i++) {
+                if ("-c".equals(args[i - 1]) || "--config".equals(args[i - 1])) {
+                    Matcher m = p.matcher(args[i]);
+                    if (m.find()) {
+                        args[i] = m.group(1);
```

Comment:
this patch seems not graceful enough:(

Reply:
@ruanwenjun Hi wenjun, in original code: [Waterdrop.java#L71](https://github.com/apache/incubator-seatunnel/blob/b1c722e3632327df3a1d65e391e5b36328c3a1a3/waterdrop-core/src/main/java/io/github/interestinglab/waterdrop/Waterdrop.java#L71), when job is spark cluster mode, path will be replaced with fileName only. in the recently code, Seatunnel.java was refactored and I can't found corresponding code, can you explain the evolution process?

Reply:
Thank you for your reply .The new version adds the parameters to the back but does not get the file name. Based on this, it can be modified by changing the parameters. Do you have any better suggestions?

Reply:
Thank you for your suggestion. Is it to extract the path and splice args again?

Reply:
I didn't test it, but I think you can use `commandArgs.getConfigFile()` to replace `Paths.get(commandArgs.getConfigFile())`, and don't need to do other change.

Reply:
@ruanwenjun Thank you for your help.i I want  to use commandArgs.getConfigFile().getFileName.but the function also need to create .so i add commits to explain  what i do .can you help me review the code and if ok approve that